src/util/file_response_builder.rs

```diff
@@ -125,8 +141,49 @@ impl FileResponseBuilder {
                 );
         }
 
+        let mut status_code = StatusCode::OK;
+        let mut file_bytes_stream = FileBytesStream::new(file);
+
+        // Compute the ranges now that we have the file size, if we're multipart then
+        // we need a boundary too.
+       
+        if let Some(ref range) = self.range {
+            if let Ok(ranges) = HttpRange::parse(&range, metadata.len()) {
```

Comment:
Maybe consider moving passing the metadata or file size to the constructor of this struct, so we can parse the ranges immediately.